debarshi/js/todo.js

```diff
@@ -0,0 +1,317 @@
+
+    if (currentTodo) {
+      const newTodo = {
+        id: new Date().getSeconds(),
+        text: inputEl.value,
+        isCompleted: false,
+      }
+      currentTodo.children.push(newTodo)
+    }
+  } else {
+    const todo = {
+      id: new Date().getSeconds(),
+      text: inputEl.value,
+      isCompleted: false,
+      children: [],
+    }
+
+    todos.push(todo)
+
+    cardContainer.classList.remove("d-none")
+    cardContainer.classList.add("padding-xs")
+    noTodoText.classList.add("d-none")
+    inputEl.value = ""
+    target.disabled = true
+
+    todoWrapper.innerHTML = ""
+  }
+  render()
+}
+
+// Clear all todos
+const clearAllTodos = () => {
+  todos.length = 0
+  todoWrapper.innerHTML = ""
+
+  cardContainer.classList.remove("padding-xs")
+  noTodoText.classList.remove("d-none")
+  cardContainer.classList.add("d-none")
+}
+
+const removeTodoNodes = () => {
+  todoWrapper.innerHTML = ""
+  render()
+}
+
+// Renders todos into the DOM
+const render = (
+  parentNode = todoWrapper,
+  items = todos,
+  draggable = true,
+  includeChildTodos = true,
+) => {
+  const docFragment = new DocumentFragment()
+
+  for (let todo of items) {
+    const todoNode = createTodoNodes(todo, draggable, includeChildTodos)
+
+    if (todo?.children?.length) {
+      parentNode.innerHTML = ""
+      render(todoNode, todo.children, false, false)
+    }
+    docFragment.appendChild(todoNode)
+  }
+
+  parentNode.appendChild(docFragment)
+}
+
+// Handle add todo button disabled status on input change
+const handleInputChange = (event) => {
+  const { value = "", parentNode } = event.target
+  const btnEl = findNode(parentNode, "BUTTON")
+
+  if (btnEl && validateInput(value)) {
+    btnEl.disabled = false
+  } else {
+    btnEl.disabled = true
+  }
+}
+
+// Marks all todos as complete
+const handleMarkAllComplete = () => {
+  const completedTodos = markTodosComplete(todos)
+  todos.length = 0
+  todos.push(...completedTodos)
+  todoWrapper.innerHTML = ""
+  render()
+}
+
+// Delegated event handler for click events
+const handleTodoClick = (event) => {
+  const { tagName, id, type } = event.target
+
+  if (tagName === "INPUT" && type === "checkbox") {
+    return toggleTodoCompleted(event)
+  } else {
+    if (tagName === "BUTTON") {
+      switch (id) {
+        case "add-sub-todo":
+          return handleAddTodo(event)
+        case "add-todo":
+          return handleAddTodo(event)
+        case "delete-todo":
+          return deleteTodo(event)
+        case "edit-todo":
+          return renderEditTodoNodes(event)
+        case "update-todo":
+          return handleEditTodo(event)
+        case "sub-todo":
+          return renderCreateSubTodoNodes(event)
+        case "cancel-sub-todo":
+          return removeTodoNodes(event)
+        case "cancel-edit-todo":
+          return removeTodoNodes(event)
+        case "clear":
+          return clearAllTodos()
+        case "mark-complete":
+          return handleMarkAllComplete()
```

Comment:
linter might give warning for not having default case. Also i suggest using object mapping (eg: {'add-sub-todo' : handleAddTodo}) instead of switch case if possible.